### DISCUSSION TITLE
Option to track Habits with no frequency

### DISCUSSION THREAD

**Discussion by drawtheliner** [iSoron/uhabits#1379] [Feature Requests]
_I understand this request might be outside of Loop's use case._

There are some habits I'd like to keep track of even though I don't really have a goal related to them. For example, sometimes I take the neighbour's dog for a walk. I don't really want/need to do this more (or less) often than I do now. Still, I'd like to track it to get some insights about it after some time.

I could just take note of this in my diary, but it would be cumbersome, unstructured and wouldn't give me any automated stats/insights. 

I guess most people would say a habit is "something that happens with a certain established frequency and time frame" (school every day, juice 3x a week, surf once a year etc.) However I believe habits can also be things that have no pattern. Taking the neighbour's dog for a walk is a habit to me, just without any clear pattern of occurrence.

Maybe ultimately this depends whether Loop is actually a habit **builder** (which would make my request out of scope), instead of a habit **tracker** (a broader use, which would encompass my request).

**Comment by LordJon** [comment]
I have made a very general habit for this which I call 'Extra tasks' that asks if I completed the extra tasks that happened that week. I stopped using it a while ago as it wasn't useful to me and I started using a todo.txt task app for this (Simpletask, also opensource).

I don't agree on habits not having a pattern though. The definition of the word 'habit' is (as far as I'm aware) a reoccurring type of behaviour that has a specific pattern.

**Reply by drawtheliner** [reply]
>  I started using a todo.txt task app for this (Simpletask, also opensource).

I'll have a look, thanks!

> I don't agree on habits not having a pattern though. The definition of the word 'habit' is (as far as I'm aware) a reoccurring type of behaviour that has a specific pattern.

Yeah, I think most people think as you. I always thought of habit as something that is recurring, but not necessarily with a fixed temporal schedule/frequency. As in "I have the habit of biting my fingernails". Also sometimes habits are not time bound, but rather contextual, as in "I have the habit of going to the park when it's warm" or  "I have the habit of wearing a hat whenever I'm sad". By the way, some habits are location based, as in "I have the habit of saying hi to the manager every time I go to The Crown pub" - this would be interesting to be implemented using location from a phone. But I digress!

Google defines "habit" as:
> a settled or regular tendency or practice, especially one that is hard to give up.

So I guess I'm talking about the "settled" part of the definition.

**Comment by hiqua** [comment]
Why do you care that there's a bogus frequency attached to your habit in this case?

**Reply by drawtheliner** [reply]
Because the app will be giving me feedback related to that bogus frequency, which would be incorrect (and distracting).

Also they get placed in bogus places when sorting, which is also confusing.

Again, I understand if this isn't a use case for the app.

**Comment by exanrauzer** [comment]
Yes, I also use the app to track events that are not habbits - it is very convenient - I have it one cool app.

For some of them I just have them being archived - and when the event happends I just view it and mark. For events which I track and which are not archived because they are quite frequent I use the most grey color in darkmode and keep them low on list. Shortly speaking I use tricks. So I would also like to not have a 'score' attached.

**Comment by hiqua** [comment]
I feel like we could possibly allow frequencies of 0 to cover this specific use-case, if that doesn't bad consequences for the rest of the app. This means that you'd get shallow checkmarks everyday.

**Reply by exanrauzer** [reply]
Yes, I imagine it may be easy to implement. How do you feel about adding a gray / colored or empty / full **dot** instead of checkmarks or crosses?

**Reply by ke352802081770314** [reply]
I do not see that a different symbol adds more clarity, to me it would be more confusing.

I wouldn't hesitate to just define such a (tracker) habit "every 365 days" which will do pretty much what was asked without requiring code adjustment. But I am not a power user, so probably I simply don't know the disadvantages this incurs (e.g. I do not sort my habits I have a manual order I created and would be very confused if this changed, no matter by what criteria)
actually I created a similar tracking habit myself "swimming" I do not have a particular target, so I entered 10 minutes per month, which hopefully I will always over-achieve. numerical habits may benefit from a more rare frequency in addition to per day/week/month, e.g. per quarter or even per year so support such use cases.

**Reply by exanrauzer** [reply]
Dots are neutral. Your swimming activity is a habit, as you have written. You'd like to do this activity. So this is not the action that would be covered by this feature.

Main goal is to not to have it 'scored', measured, and to distinguish it from a habit.

**Reply by ke352802081770314** [reply]
I called it a "(tracker) habit" to distinguish from a "goal habit" while keeping the term habit which is the name of items in this app, consistent with its name Habit Tracker. Also the subject of this suggestion is "Option to track **Habits** with no frequency" so I find it logical to stick to the term chosen by the initiator.
Please stay constructive instead of nitpicking.

**Comment by herrbrixter** [comment]
I also want this. I track this for occurences of something.

How many times my dandruff flared up
How many times an exciting thing happened

I just want to track when it happens without having a goal.

**Comment by GutHib** [comment]
Maybe a compromise would be to add a "per year" setting. That would still only be a workaround, but the workaround would be so generous that "take the dog for a walk at least once a year" would almost be equivalent to not setting a goal at all.
